hw_diag/views/diagnostics.py

```diff
@@ -0,0 +1,53 @@
+import json
+import base64
+
+from flask import Blueprint
+from flask import render_template
+from flask import request
+from flask import jsonify
+
+
+DIAGNOSTICS = Blueprint('DIAGNOSTICS', __name__)
+
+
+def read_diagnostics_file():
+    diagnostics = {}
+    try:
+        with open('diagnostic_data.json', 'r') as f:
+            diagnostics = json.load(f)
+    except FileNotFoundError:
+        msg = 'Diagnostics have not yet run, please try again in a few minutes'
+        diagnostics = {'error': msg}
+    return diagnostics
+
+
+@DIAGNOSTICS.route('/')
+def get_diagnostics():
+    diagnostics = read_diagnostics_file()
+
+    if request.args.get('json'):
+        return jsonify(diagnostics)
+
+    return render_template('diagnostics_page.html', diagnostics=diagnostics)
+
+
+@DIAGNOSTICS.route('/initFile.txt')
+def get_initialisation_file():
+    diagnostics = read_diagnostics_file()
+
+    if diagnostics.get('error'):
+        return 'Internal Server Error', 500
+
+    response = {
```

Comment:
Why do we return a subset of all information?

Reply:
A requirement was to retain backwards compatibility with the old app, this is used to generate /initFile.txt and is used in the factory for device onboarding.